google/fingerprinters/web/scripts/updater/community/spark/update.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+source ../../common.sh
+
+SCRIPT_PATH="$(cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P)"
+# Root path to the web fingerprinter plugin.
+PROJECT_ROOT="$(cd -- "${SCRIPT_PATH}/../../../.." >/dev/null 2>&1 ; pwd -P)"
+# Path to the configurations for starting a live instance of Spark.
+SPARK_APP_PATH="${SCRIPT_PATH}/app"
+# Path to the temporary data holder.
+TMP_DATA="/tmp/SPARK_fingerprints"
+# Path to the local git repository for Spark codebase.
+GIT_REPO="${TMP_DATA}/repo"
+# Path to the directory of all the updated fingerprints data.
+FINGERPRINTS_PATH="${TMP_DATA}/fingerprints"
+# Json data of the final result.
+JSON_DATA="${FINGERPRINTS_PATH}/fingerprint.json"
+# Binary proto data of the final result.
+BIN_DATA="${FINGERPRINTS_PATH}/fingerprint.binproto"
+# Read all the versions to be fingerprinted.
+readarray -t ALL_VERSIONS < "${SCRIPT_PATH}/versions.txt"
+mkdir -p "${FINGERPRINTS_PATH}"
+
+startSpark() {
+  local version="$1"
+  pushd "${SPARK_APP_PATH}" >/dev/null
+    # if version-python3 exists then we have a spark container with python3
+    # otherwise we must install python3
+    $(SPARK_VERSION="${version}-python3" docker compose up -d &&
+        sleep 10) || \
+      (
+        echo -e "\nInstalling python3 into worker container"
+        SPARK_VERSION="${version}" docker compose up -d
+        sleep 10
+        installPython3InSpark "${version}"
+      )
+  popd >/dev/null
+}
+
+installPython3InSpark() {
+  local version="$1"
+  pushd "${SPARK_APP_PATH}" >/dev/null
+    docker exec  -it -u 0 spark-master apt update >/dev/null
+    docker exec  -it -u 0 spark-master apt install python3 python3-pip -y >/dev/null
+    docker exec  -it -u 0 spark-master pip3 install pyspark=="${version}" >/dev/null
```

Comment:
```suggestion
    docker exec -it -u 0 spark-master apt-get update >/dev/null
    docker exec -it -u 0 spark-master apt-get install python3 python3-pip -y >/dev/null
    docker exec -it -u 0 spark-master pip3 install pyspark=="${version}" >/dev/null
```
`apt` is for interactive use, while `apt-get` is better suited for scripts